examples/s1_generic.py

```diff
@@ -0,0 +1,309 @@
+import argparse
+import collections
+import datetime
+import io
+import logging
+import secrets
+import time
+import urllib.parse
+from http import HTTPStatus
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+import requests
+import requests.adapters
+from intezer_sdk import api
+from intezer_sdk import errors
+from intezer_sdk.analysis import Analysis
+
+_s1_session: Optional[requests.Session] = None
+_logger = logging.getLogger('intezer')
+
+
+class BaseUrlSession(requests.Session):
+    base_url = None
+
+    def __init__(self, base_url=None):
+        if base_url:
+            self.base_url = base_url
+        super(BaseUrlSession, self).__init__()
+
+    def request(self, method, url, *args, **kwargs):
+        'Send the request after generating the complete URL.'
+        url = self.create_url(url)
+        return super(BaseUrlSession, self).request(
+            method, url, *args, **kwargs
+        )
+
+    def prepare_request(self, request):
+        'Prepare the request after generating the complete URL.'
+        request.url = self.create_url(request.url)
+        return super(BaseUrlSession, self).prepare_request(request)
+
+    def create_url(self, url):
+        'Create the URL based off this partial path.'
+        return urllib.parse.urljoin(self.base_url, url)
+
+
+def init_s1_requests_session(api_token: str, base_url: str):
+    headers = {'Authorization': 'ApiToken ' + api_token}
+    global _s1_session
+    _s1_session = BaseUrlSession(base_url)
+    _s1_session.headers = headers
+    # _s1_session.verify = False  # TODO
+    _s1_session.mount('https://', requests.adapters.HTTPAdapter(max_retries=3))
+    _s1_session.mount('http://', requests.adapters.HTTPAdapter(max_retries=3))
+
+
+def analyze_by_file(threat_id: str):
+    download_url, zipp_password = fetch_file(threat_id)
+    file = download_file(download_url)
+    analysis = Analysis(file_stream=file, file_name=f'{threat_id}.zip', zip_password=zipp_password)
+    return analysis
+
+
+def fetch_file(threat_id: str) -> Tuple[str, Optional[str]]:
+    zip_password = secrets.token_urlsafe(32)
+    fetch_file_time = datetime.datetime.utcnow() - datetime.timedelta(seconds=5)
+
+    response = _s1_session.post('/web/api/v2.1/threats/fetch-file',
+                                json={'data': {'password': zip_password}, 'filter': {'ids': [threat_id]}})
+    assert_s1_response(response)
+
+    for _ in range(20):
+        time.sleep(10)
```

Comment:
i would add a debug log here with the counter

Reply:
Done